emukit/quadrature/interfaces/standard_kernels.py

```diff
@@ -84,6 +84,67 @@ def dKdiag_dx(self, x: np.ndarray) -> np.ndarray:
         return np.zeros((x.shape[1], x.shape[0]))
 
 
+class IProductMatern12(IStandardKernel):
+    r"""Interface for a Matern12 (a.k.a. Exponential) product kernel.
+
+    The product kernel is of the form
+    :math:`k(x, x') = \sigma^2 \prod_{i=1}^d k_i(x, x')` where
+
+    .. math::
+        k_i(x, x') = e^{-r_i}.
+
+    :math:`d` is the input dimensionality,
+    :math:`r_i:=\frac{|x_i - x'_i|}{\lambda_i}`,
+    :math:`\sigma^2` is the ``variance`` property and :math:`\lambda_i` is the :math:`i` th element
+    of the ``lengthscales`` property.
+
+    Make sure to encode only a single variance parameter, and not one for each individual :math:`k_i`.
+
+    .. note::
+        Inherit from this class to wrap your standard product Matern12 kernel. The wrapped kernel can then be
+        handed to a quadrature product Matern12 kernel that augments it with integrability.
+
+    .. seealso::
+       * :class:`emukit.quadrature.kernels.QuadratureProductMatern12`
+       * :class:`emukit.quadrature.kernels.QuadratureProductMatern12LebesgueMeasure`
+
+    """
+
+    @property
+    def nu(self) -> float:
+        """The smoothness parameter of the kernel."""
+        return 0.5
+
+    @property
+    def lengthscales(self) -> np.ndarray:
+        r"""The lengthscales :math:`\lambda` of the kernel."""
+        raise NotImplementedError
+
+    @property
+    def variance(self) -> float:
+        r"""The scale :math:`\sigma^2` of the kernel."""
+        raise NotImplementedError
+
+    def _dK_dx1_1d(self, x1: np.ndarray, x2: np.ndarray, ell: float) -> np.ndarray:
+        """Unscaled gradient of 1D Matern12 where ``ell`` is the lengthscale parameter.
```

Comment:
`ell` sounds really cryptic. WHy not just `lengthscales`?

Reply:
All product kernels in the `quadrature` package have methods that act on the d-dimensional space and other methods that solely act on an individual dimension of the product kernel. In order to make it clear if a method acts on all or a single dimension I used `lengthscales` and `ell` respectively (`ell` is from LaTeX's \ell command) inside the methods. I found `lengthscale` to be too similar to `lengthscales` that it confused me. 

I changed the key from "ell" to "lengthscale" in the dictionaries that define the per-dimension parameters for all the product kernels, so it's a bit more clear what `ell` is supposed to be. But let me know if it's still too confusing, then I'll change it.

Reply:
Still very cryptic. Perhaps you could do something nice with prefixes? E.g. `full_space_lengthscale` and `single_dim_lengthscale` or something of that sort?

It actually took me a while to realise that `ell` isn't an abbreviation at all. Shame on me for sure, but also kinda highlights how confusing it can be for some

Reply:
Probably `lengthscales` should stay as is as it's part of the interface and it's also similar to the interface of GPy kernels. I'll change `ell` to `lengthscale` then as you originally proposed. If the variable names get too long in algebraic expressions then the code is also hard to read which may be even more confusing, especially with all the black formatting. Sounds good?

I'll do it in a quick follow up PR though if that's OK as I need to change this in all product kernels and so the changes are a bit better separated from this PR.